syncer/syncer.go

```diff
@@ -1360,6 +1399,9 @@ func (s *Syncer) syncDML(
 		case sqlJob, ok := <-jobChan:
 			metrics.QueueSizeGauge.WithLabelValues(s.cfg.Name, queueBucket, s.cfg.SourceID).Set(float64(len(jobChan)))
 			if !ok {
+				if len(jobs) > 0 {
+					tctx.L().Warn("have unexecuted jobs when close job chan!", zap.Any("rest job", jobs))
+				}
```

Comment:
Why do we directly return here instead of executing the remained SQLs?

Reply:
This is the same logic as before. I prefer to execute the remaining jobs before return. Though currently, the caller can guarantee that all messages have been processed before the chan is closed.

Reply:
Because the job in jobQueue have been executed before `closeJobChan()`. This is because subtask's(also syncer's) ctx must cancel before `closeJobChan()`, and that means all jobs will be executed before exit `syncer.Run()`.

Reply:
It's a little tricky here but SGTM.